service-loadbalancer/service_loadbalancer.go

```diff
@@ -367,6 +377,14 @@ func (lbc *loadBalancerController) getServices() (httpSvc []service, tcpSvc []se
 			glog.Infof("Ignoring service %v, it already has a loadbalancer", s.Name)
 			continue
 		}
+
+		if *nameLabel != "" {
+			if _, ok := s.Labels[*nameLabel]; !ok {
```

Comment:
can you avoid doing this in 2 places by having getServiceLB return an error and skipping? or move it out of that method and keep all the lbc.nameLabel logic here.